src/libtiled/minimaprenderer.cpp

```diff
@@ -203,6 +203,8 @@ void MiniMapRenderer::renderToImage(QImage &image, RenderFlags renderFlags) cons
                             painter.translate(-origin);
                         }
 
+                        painter.setOpacity(object->opacity());
```

Comment:
We'll need to store the above `layer->effectiveOpacity()` in a variable and set the opacity to `effectiveOpacity * object->opacity()` here.

Reply:
Wow. I don't think I would have even considered that. This is one of my fears of being a contributor. I do not know the full requirements for such a project. I am just learning qt to be honest.